### DISCUSSION TITLE
RPi4 8GB

### DISCUSSION THREAD

**Discussion by GaryLittlemore** [veryengineer/collab#20] [Q&A]
So I've got x3 RPi4 8GB on order, looking to use 2 as nodes.

What's the best way at the moment to implement them?

**Comment by DogestMaximus** [comment]
Nice; 8G Rpi4 is amazing I have one myself and they download the blockchain fairly quickly.  As far as appliance it's entirely up to you; we are aiming to provide the solutions for all of these major architectures but as far as what you do with your device it's your choice really.  The 8G Rpi 4 is x64 so it could even serve as a working and usable computer.

I'm programming some of my SBC's into open-source home-automation which if it works correctly should give you a ton of options; others have had some great ideas in this thread so it's really who wants to build whatever- I think we plan on providing base images and maybe even a rollout or two of something specific if enough people ask for it.  The primary objective is network support and nodes with reliable hardware- so absolutely anything you can do.

I can test out an Rpi4 image when I get a little further it's def on the todo.

Thanks for your interest and help with the support; check in periodically and we'll have something or if you have an idea feel free to start a discussion.

**Reply by GaryLittlemore** [reply]
I gave up on my RPi 3b+ it kept freezing trying to get the blocks in. So I'll try again once the RPi 4 arrives.

**Reply by DogestMaximus** [reply]
We were working on this a few days ago and I believe it's been pushed: big thanks to [Xanimo](https://github.com/xanimo) @[KBluezr](https://twitter.com/KBluezr) for working hard on aarch64 and Qt.

We noticed it to on the newer hardware and spent some time compiling it to find out why.

**Reply by patricklodder** [reply]
> I gave up on my RPi 3b+ it kept freezing trying to get the blocks in. So I'll try again once the RPi 4 arrives.

Do you have a `debug.log` and a `journalctl` dump for that? Could be useful to see if we can determine if that is a hardware, OS or software issue.

**Comment by DogestMaximus** [comment]
The Rpi3 and 3 B+ have some networking limitations in the form of bottlenecks in their hardware design; combine that with only 1G of RAM (minimum SWAP of 2G) and microSD selection there's a variety of factors that can attribute  as to "why" in my opinion and I don't believe it's software related at all.  For testing it served a purpose- but I don't ever see it being a long-term solution.

One of the things that may have made a difference as far as success was it's ability to boot and store the Blockchain on USB.
